controllers/articles.js

```diff
@@ -0,0 +1,195 @@
+        article: {
+          title: updateThisArticle.title,
+          body: updateThisArticle.body,
+          slug: updateThisArticle.slug,
+        }
+      });
+    } catch (err) {
+      if (err.message) {
+        res.status(500).send({
+          status: res.statusCode,
+          errorMessage: 'No article to update, please create an article first',
+        });
+      }
+    }
+  },
+
+  /*
+   * Deleting a post based to its slug.
+   *
+   * For avoiding permanent deletion of the content from the database
+   * we set the flag to the article that it is deletes.
+   */
+  deleteArticle: async (req, res) => {
+    const { slug } = req.params;
+    const row = await article.update({ deleted: 1 }, { where: { slug, deleted: 0 } });
+    if (row[0] === 0) {
+      return res.status(404).send({
+        status: res.statusCode,
+        message: 'No article found for this slug',
+      });
+    }
+    return res.status(200).send({
+      status: res.statusCode,
+      message: 'Article deleted successfully',
+    });
+  },
+
+  /*
+  * Retieving all the published articles based to the author
+  * and the status of the article (Published).
+  */
+  getAllPublished: async (req, res) => {
+    try {
+      // const { author } = req.user;
+      const response = await article.findAll({ where: { author: tempUser, status: 'Published', deleted: 0 } });
+      return res.status(200).send({
+        status: res.statusCode,
+        articles: response,
+      });
+    } catch (error) {
+      throw error;
+    }
+  },
+
+  /*
+  * Retieving all the published articles based to the author
+  * and the status of the article (Draft).
+  */
+  getAllDraft: async (req, res) => {
+    // const { author } = req.user;
+    try {
+      const response = await article.findAll({ where: { author: tempUser, status: 'Draft', deleted: 0 } });
+      return res.status(200).send({
+        status: res.statusCode,
+        articles: response,
+      });
+    } catch (error) {
+      throw error;
+    }
+  },
+
+  /*
+   * Construct user's feed
+   * and the status of the article (Published).
+   */
+  getFeeds: async (req, res) => {
+    try {
+      const response = await article.findAll({
+        where: {
+          status: 'Published',
+          deleted: 0
+        }
+      });
+      return res.status(200).send({
+        status: res.statusCode,
+        articles: response,
+      });
+    } catch (error) {
+      throw error;
+    }
+  },
+
+  viewAnArticle: async (req, res) => {
+    const { slug } = req.params;
+    try {
+      const findAuthor = await User.findOne();
+      const viewOneArticle = await article.findOne({ where: { slug } });
+      if (!viewOneArticle && !findAuthor) {
+        return res.status(404).send({
+          status: res.statusCode,
+          errorMessage: 'No article found, please create an article first'
+        });
+      }
+      return res.status(200).send({
+        status: res.statusCode,
+        article: {
```

Comment:
I think you should also return a tag list, as when creating a story you saved tags?